### DISCUSSION TITLE
read stale value from a "derived" proxy

### DISCUSSION THREAD

**Discussion by xiechao06** [pmndrs/valtio#618] [Q&A]
## Summary

mix-use of  *useState* and *valtio.derive* will make the component read the stale value of the derived proxy.

## Link to reproduction

https://codesandbox.io/s/valtio-mix-with-state-ielduu?file=/src/App.tsx

Though this example seems harmless, this inconsistency brokes my project (in the progress of migration from redux to valtio).

but it works fine if use valtio only, https://codesandbox.io/s/valtio-not-mix-with-state-2j4ocl?file=/src/App.tsx

BTW, I've tested jotai, it also works, https://codesandbox.io/s/brave-drake-qsvvz5?file=/src/App.tsx, would you suggest using jotai instead?

**Comment by dai-shi** [comment]
This is a tricky one. `derive` will delay a little bit on purpose.
It can be optionally be in `sync`.
Check this: https://codesandbox.io/s/valtio-mix-with-state-forked-3ik7n3

`proxyWithComputed` will also work in this case.

> would you suggest using jotai instead?

Well, it depends. To my understanding, usually people who pick Valtio like its mutable syntax which Jotai can't do.
